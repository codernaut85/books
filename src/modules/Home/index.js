import React from 'react';
import { Link } from "react-router-dom";

function Home() {
  return <section className="wrapper">
    <h1>Homepage</h1>
    <p>Welcome to the <Link className="link" to="/books/page/1">book archive</Link></p>
  </section>
};

export default Home;