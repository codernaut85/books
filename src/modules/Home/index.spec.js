import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';
import Home from './index';

const mockStore = configureStore([]);

describe("Home", () => {  
  it("renders the simple home screen correctly", () => {
    const store = mockStore({
      books: {
        books: [],
        count: 0
      }
    });

    const testRenderer = TestRenderer.create(<Router>
      <Provider store={store}>
        <Home />
      </Provider>
    </Router>);
    
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});