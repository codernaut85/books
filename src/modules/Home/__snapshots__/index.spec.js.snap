// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the simple home screen correctly 1`] = `
<section
  className="wrapper"
>
  <h1>
    Homepage
  </h1>
  <p>
    Welcome to the 
    <a
      className="link"
      href="/books/page/1"
      onClick={[Function]}
    >
      book archive
    </a>
  </p>
</section>
`;
